src/Microsoft.OpenApi.Hidi/OpenApiService.cs

```diff
@@ -110,43 +111,13 @@ CancellationToken cancellationToken
                 else
                 {
                     stream = await GetStream(openapi, logger, cancellationToken);
-
-                    using (logger.BeginScope($"Parse OpenAPI: {openapi}",openapi))
-                    {
-                        stopwatch.Restart();
-                        var result = await new OpenApiStreamReader(new OpenApiReaderSettings
-                        {
-                            RuleSet = ValidationRuleSet.GetDefaultRuleSet(),
-                            LoadExternalRefs = inlineExternal,
-                            BaseUrl = openapi.StartsWith("http") ? new Uri(openapi) : new Uri("file:" + new FileInfo(openapi).DirectoryName + "\\")
-                        }
-                        ).ReadAsync(stream);
-
-                        document = result.OpenApiDocument;
-
-                        var context = result.OpenApiDiagnostic;
-                        if (context.Errors.Count > 0)
-                        {
-                            logger.LogTrace("{timestamp}ms: Parsed OpenAPI with errors. {count} paths found.", stopwatch.ElapsedMilliseconds, document.Paths.Count);
-
-                            var errorReport = new StringBuilder();
-
-                            foreach (var error in context.Errors)
-                            {
-                                logger.LogError("OpenApi Parsing error: {message}", error.ToString());
-                                errorReport.AppendLine(error.ToString());
-                            }
-                            logger.LogError($"{stopwatch.ElapsedMilliseconds}ms: OpenApi Parsing errors {string.Join(Environment.NewLine, context.Errors.Select(e => e.Message).ToArray())}");
-                        }
-                        else
-                        {
-                            logger.LogTrace("{timestamp}ms: Parsed OpenApi successfully. {count} paths found.", stopwatch.ElapsedMilliseconds, document.Paths.Count);
-                        }
-
-                        openApiFormat = format ?? GetOpenApiFormat(openapi, logger);
-                        openApiVersion = version != null ? TryParseOpenApiSpecVersion(version) : result.OpenApiDiagnostic.SpecificationVersion;
-                        stopwatch.Stop();
-                    }
+                    stopwatch.Restart();
+                    var result = await ParseOpenApi(openapi, logger, stream);
```

Comment:
The `TransformOpenApiDocument()` method uses these additional custom OpenApiReaderSettings in the OpenApiStreamReader which are missing in the new `ParseOpenApi()` method. Does this mean we no longer need these settings?
`LoadExternalRefs = inlineExternal`
`BaseUrl = openapi.StartsWith("http") ? new Uri(openapi) : new Uri("file:" + new FileInfo(openapi).DirectoryName + "\\")`

Reply:
My intent was that the behavior would be identical. I will review what settings got lost.